vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/AuthenticationRequest.kt

```diff
@@ -1,16 +1,62 @@
 package at.asitplus.wallet.lib.oidc
 
-import io.ktor.http.*
+import at.asitplus.KmmResult.Companion.wrap
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
 
-sealed class AuthenticationRequestParametersFrom<T>(val source: T, val parameters: AuthenticationRequestParameters) {
-    class JwsSigned(
-        jwsSigned: at.asitplus.crypto.datatypes.jws.JwsSigned,
-        parameters: AuthenticationRequestParameters
-    ) : AuthenticationRequestParametersFrom<at.asitplus.crypto.datatypes.jws.JwsSigned>(jwsSigned, parameters)
+/**
+ * Helper class for allowing serialization of AuthenticationRequestParametersFrom<*>
+ */
+// TODO: possibly replace this with AuthenticationRequestParametersFrom<*> once it's serializable? or the other way?
```

Comment:
Maybe two things are being mingled together that should be separate, or we need to revamp AuthenticationReuqestParametersFrom. Currently, AuthenticationRequestParametersFrom contains the source of said parameters. so the serialized version of the parameters is present. It is not meant to be used to construct such parameters from scratch. While everything I see here looks like it is perfectly fine to merge both classes together. My gut feeling tells me that this might backfire. I am undecided, as I can't really put my finger on the reason I'm reluctant.

Reply:
Yeah, somehow this seems mixed up ... otherwise, if the class is now simply serializable, why not?

Reply:
My understanding is, that the main purpose of the `source` parameter of AuthenticationReuqestParametersFrom is to retain request metadata (eg. jws header) for further validation, or possibly for error reporting, right?

Is this an issue that blocks the pull request from merging?

Reply:
yes that's correct. So can you or can you not merge both classes together so that everything still makes sense?
If it is **mostly** the same, then a common base class for both makes sense, with one serializable class being used for creation.
The benefit of this strategy over leaving a TODO and an issue is that if we discover that this class hiararchy collapses into a single class, we can just use the refactoring tools of the IDE and get rid of in within 5 minutes, while not adding any redundancies to the code.

Reply:
> So can you or can you not merge both classes together so that everything still makes sense

I didn't realize this was a question up until now, but yes, the class was intended to hold the same members;
done in commit 81a3815eda17e6e9cd56e40216bb271092ef2773

Reply:
given all checks are passing, I'd say it is perfectly fine like this